setup/src/Magento/Setup/Test/Unit/Module/Di/Code/Reader/FileClassScannerTest.php

```diff
@@ -117,42 +117,15 @@ public function test()
         self::assertCount(1, $result);
         self::assertContains('This\Is\My\Namespace\ThisIsMyTest', $result);
     }
-    public function testGetMultipleClassesInMultiNamespace()
-    {
-        $scanner = $this->getMockBuilder(FileClassScanner::class)->disableOriginalConstructor()->setMethods([
-            'getFileContents'
-        ])->getMock();
-        $scanner->expects(self::once())->method('getFileContents')->willReturn(<<<PHP
-<?php
-
-namespace This\Is\My\Namespace;
-
-class ThisIsMyTest {
 
-}
-
-class ThisIsAnotherTest {
-
-    public function __construct()
+    public function testTheWeirdExceptionCaseThatHappens()
     {
-    
-    }
-    
-    public function test()
-    {
-        return self::class;
-    }
-
-}
-PHP
-        );
-        /* @var $scanner FileClassScanner */
-
+        $filename = __DIR__ . '/../../../../../../../../../../app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Eav/AbstractEav.php';
```

Comment:
It is similar to #8782.  The class keyword has the same token number as the class definition keyword.  That means that the code used to determine the class name has to understand the different contexts.  I had tried to write the test so that I could mock the code but I couldn't get it to replicate.  So I just referenced the one file that was breaking.